File: Problems/0901.-1000./0941._Valid_Mountain_Array/ValidMountainArray1.py
#!/bin/python3
class Solution(object):
    def validMountainArray(self, arr: list) -> bool:
        i, j = 0, len(arr)-1
        # Strictly INCREASING loop
        while (i < len(arr)-1 and arr[i] < arr[i+1]):
            i += 1
        # Strictly DECREASING loop
        while (j > 0 and arr[j-1] > arr[j]):
            j -= 1
        '''
        If the array is a VALID MOUNTAIN, i and j meet
        at the same index
        '''
        return (i > 0 and i == j and j < len(arr)-1)

File: Problems/0901.-1000./0941._Valid_Mountain_Array/ValidMountainArray1.java
//!/bin/java
class Solution {
    public boolean validMountainArray(int[] arr) {
        int n = arr.length, i = 0, j = n-1;

        // Strictly INCREASING loop
        while (i < n-1 && arr[i] < arr[i+1])
            i++;

        // Strictly DECREASING loop
        while (j > 0 && arr[j-1] > arr[j])
            j--;

        /*
         * If the array is a VALID MOUNTAIN array,  i and j meet
         * at the same index
         */
        return (i > 0 && i == j && j < n-1);
    }
}
